mobilecoind/src/service.rs

```diff
@@ -537,11 +537,32 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
             rpc_internal_error("ledger_db.get_tx_out_by_index", err, &self.logger)
         })?;
 
-        // Use root entropy to construct AccountKey.
-        let mut root_entropy = [0u8; 32];
-        root_entropy.copy_from_slice(transfer_payload.get_root_entropy());
-        let root_id = RootIdentity::from(&root_entropy);
-        let account_key = AccountKey::from(&root_id);
+        // Use bip39 or root entropy to construct AccountKey.
+        let account_key = if !transfer_payload.get_bip39_entropy().is_empty() {
+            let mut bip39_entropy = [0u8; 32];
+            if bip39_entropy.len() != transfer_payload.get_bip39_entropy().len() {
+                return Err(RpcStatus::new(
+                    RpcStatusCode::INVALID_ARGUMENT,
+                    Some("bip39_entropy".to_string()),
+                ));
+            }
+            bip39_entropy.copy_from_slice(transfer_payload.get_bip39_entropy());
+            let mnemonic = Mnemonic::from_entropy(&bip39_entropy, Language::English)
+                .map_err(|err| rpc_internal_error("Mnemonic.from_entropy", err, &self.logger))?;
+            let key = mnemonic.derive_slip10_key(0);
```

Comment:
I'm reading this to mean that transfer codes must always use an `account_index` of `0`. If so, we should make sure that's communicated elsewhere as well.

Reply:
That is correct. Where do you think we should communicate this? I don't see any reason to support different `account_index`es for this since the entropy/account is for a one-time use.

Reply:
Mentioning it in `printable.proto`, as part of the doc comment for the new `bip39_entropy` seems good.

Reply:
Done